packages/realm_dart/test/sync_migration_test.dart

```diff
@@ -0,0 +1,259 @@
+// Copyright 2022 MongoDB, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import 'dart:typed_data';
+
+import 'package:realm_dart/realm.dart';
+import 'package:realm_dart/src/configuration.dart';
+import 'package:test/test.dart' hide test, throws;
+
+import 'test.dart';
+
+part 'sync_migration_test.realm.dart';
+
+// This has to match the server-side schema declared in baas_client::_nullablesSchemaV0
+@RealmModel()
+@MapTo("Nullables")
+class _NullablesV0 {
+  @PrimaryKey()
+  @MapTo('_id')
+  late ObjectId id;
+
+  late ObjectId differentiator;
+
+  late bool? boolValue;
+  late int? intValue;
+  late double? doubleValue;
+  late Decimal128? decimalValue;
+  late DateTime? dateValue;
+  late String? stringValue;
+  late ObjectId? objectIdValue;
+  late Uuid? uuidValue;
+  late Uint8List? binaryValue;
+}
+
+// This has to match the server-side schema declared in baas_client::_nullablesSchemaV1
+@RealmModel()
+@MapTo("Nullables")
+class _NullablesV1 {
```

Comment:
I find this to be a bit confusing name. Would 
`_V0SchemaWithNullables` and `_V1SchemaWithoutNullables` perhaps be better.